### ISSUE TITLE
WslRegisterDistribution failed with error: 0x80072746 [Updated with a Log]

### ISSUE THREAD

**Issue by silver2row** [microsoft/WSL#11818]
### Windows Version

Windows version: 10.0.22635.3930

### WSL Version

WSL version: 2.3.11.0

### Are you using WSL 1 or WSL 2?

- [X] WSL 2
- [ ] WSL 1

### Kernel Version

Kernel version: 6.6.36.3-1

### Distro Version

Debian

### Repro Steps

`wsl --install Debian`

### Expected Behavior

I was expecting on Debian to be installed in my WSL2 instance.

### Actual Behavior

```
wsl --install Debian

Debian GNU/Linux is already installed.

Launching Debian GNU/Linux...

Installing, this may take a few minutes...

WslRegisterDistribution failed with error: 0x80072746

Error: 0x80072746 An existing connection was forcibly closed by the remote host.

Press any key to continue...

The operation completed successfully.

Then, when I open Debian after the install, the output is this:

Installing, this may take a few minutes...

WslRegisterDistribution failed with error: 0x80072749

Error: 0x80072749 A request to send or receive data was disallowed because the socket is not connected and (when sending on a datagram socket using a sendto call) no address was supplied.

Press any key to continue...
```

After I press any key, the Distro WSL2 instance of Debian disappears.

**Comment by silver2row** [comment]
No sir-re-Bob, those listed 80% closeness rated hyperlinks serve no help to me. I tried some ideas in those links. Notta.

Seth

**Comment by WenjieXu2000** [comment]
Just a few days ago, after WSL automatically updated to version 2.3.11, I encountered the same issue as you. Rolling back to version 2.2.4 didn't resolve it either, and I couldn't use any WSL distributions. However, I discovered that this issue affects WSL 2 specifically. I was able to install Linux distributions with WSL 1.

**Comment by silver2row** [comment]
Right, my linux dev. desktop holds my priceless fans.

**Comment by silver2row** [comment]
@WenjieXu2000 ,

Hello...thank you for your support. Is this just a common error that cannot be solved or are people, behind the scenes, working to address this issue as a viable error?

Seth

P.S. I want to keep WSL2 for all it offers but if WSL1 is the only cure for the error at hand, I can revert.

**Comment by silver2row** [comment]
@WenjieXu2000 , it worked. Thank you. I will test this for now.

Seth

**Comment by silver2row** [comment]
Does WSL2 work at all these days?

**Comment by silver2row** [comment]
I mean...I can get kernel 4.4.x to work via Debian on WSL1 but I miss my advancements in programming and newer language skills on VS Code and WSL2. 

If you guys get it to work, please let me know. 

Seth

**Comment by WenjieXu2000** [comment]
@silver2row, I've tried various reset methods, including removing WSL, Hyper-V, and Virtual Machine Platform, as well as disabling and re-enabling virtualization, but nothing has worked.

What's puzzling is that both of my computers have the latest version of WSL, yet only one of them is experiencing this problem and I can't seem to fix it!

**Comment by silver2row** [comment]
Hey @WenjieXu2000 ,

I think someone or "Me" erased an environment variable that may have something to do with the correct adaptation of WSL2 on my current working desktop. 

Seth

P.S. In environment variables, maybe check the one that is workin' and port the correct variable to the computer that is not working. 

```
Another Note:
```

I noticed that my current working desktop is going through issues on screen. The taskbar, hopefully totally unrelated, is starting to miss icons...

Anyway, I will get a working computer one day and look up the environment variables needed to have WSL2 instances run on it.

**Comment by silver2row** [comment]
So, something like:

```
dir env:
```

That will list your environment variables on your current desktop. I have WSLENV listed but it is not in working condition right now.

Seth

**Comment by silver2row** [comment]
https://devblogs.microsoft.com/commandline/share-environment-vars-between-wsl-and-windows/

That may help. 

Seth

P.S. It is not a command of sorts but an understanding of how WSLENV works with WSL(N) and Windows(N). I cannot find more right now.

**Comment by silver2row** [comment]
https://gist.github.com/silver2row/8da631393f945366ed13d68535b05b00

**Comment by silver2row** [comment]
[WslLogs-2024-07-25_00-13-54.zip](https://github.com/user-attachments/files/16370803/WslLogs-2024-07-25_00-13-54.zip)

**Comment by silver2row** [comment]
What now?

**Comment by WenjieXu2000** [comment]
@silver2row 
Hi, I just rolled back to WSL version 2.2.4, and it seems to have resolved the issue. I successfully installed Debian and started it. Unlike before, this time I reinstalled using the wsl.2.2.4.0.x64.msi package instead of Microsoft.WSL_2.2.4.0_x64_ARM64.msixbundle.

Now, I've upgraded back to version 2.3.12, and the previous issue has reappeared. You might want to try rolling back to the wsl.2.2.4.0.x64.msi version of WSL to see if it resolves the issue.

**Comment by silver2row** [comment]
wsl.2.2.4.0.x64 and right. That worked.

Seth

P.S. Thank you.

**Comment by silver2row** [comment]
I installed the latest kernel and had a crash. Oops.

**Comment by silver2row** [comment]
Sorry...too embarrassed.

**Comment by zxy719** [comment]
> @silver2row Hi, I just rolled back to WSL version 2.2.4, and it seems to have resolved the issue. I successfully installed Debian and started it. Unlike before, this time I reinstalled using the wsl.2.2.4.0.x64.msi package instead of Microsoft.WSL_2.2.4.0_x64_ARM64.msixbundle.您好，我刚刚回滚到 WSL 版本 2.2.4，似乎已经解决了该问题。我成功安装了 Debian 并启动了它。与以前不同，这次我使用 wsl.2.2.4.0.x64.msi 包而不是 Microsoft.WSL_2.2.4.0_x64_ARM64.msixbundle 重新安装。
> 
> Now, I've upgraded back to version 2.3.12, and the previous issue has reappeared. You might want to try rolling back to the wsl.2.2.4.0.x64.msi version of WSL to see if it resolves the issue.现在，我已经升级回2.3.12版本了，之前的问题又出现了。您可能需要尝试回滚到 WSL 的 wsl.2.2.4.0.x64.msi 版本以查看是否可以解决问题。

Thank you so much! I tried many methods without any progress until I found your reply.

**Comment by serunati** [comment]
I searched on this a while and did not find an answer; however, I stumbled upon what has fixed my occurrence of this problem in WDL 2.0.

My fix was to remove the 'localhostForwarding' configuration from my ~/.wslconfig

TLDR:
If you have this or a similar issue - strip your windows .wslconfig down to minimums and it will likely correct the problem.

~/.wslconfig
[wsl2]
memory=6GB
processors=4

**Comment by KevinLongProfessional** [comment]
I had this issue and was able to get it to work again after deleting the .wslconfig file from C:/Users/USERNAME_HERE.
